extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionScopedNotifier.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.narayana.jta.runtime;
+
+import javax.enterprise.context.BeforeDestroyed;
+import javax.enterprise.context.Destroyed;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Event;
+import javax.transaction.TransactionScoped;
+
+import io.quarkus.arc.Arc;
+
+abstract class TransactionScopedNotifier {
+
+    private transient Event<Object> initialized;
+    private transient Event<Object> beforeDestroyed;
+    private transient Event<Object> destroyed;
+
+    void initialized(Object payload) {
```

Comment:
To stay aligned with what Narayana does internally, the argument here should be `Transaction`

Reply:
Well, we reuse this component for both TM and UT so it can't be `Transaction`. That said, we could make this class a parameterized type and use `Transaction` for TM and `UserTransaction` for UT but it's IMO not worth the effort because the runtime type of the payload is used to resolve the observer methods anyway. So an `@Observes Transaction` should still be notified.

Reply:
> because the runtime type of the payload is used to resolve the observer methods anyway

Good point, then it's fine as-is